tests/sentry/services/test_organization_actions.py

```diff
@@ -260,3 +260,19 @@ def test_slug_above_size_limit(self):
         )
         assert len(slug) == 30
         assert slug == "areallylongsentryorg-945bda148"
+
+    def test_slug_with_mixed_casing(self):
+        slug = generate_deterministic_organization_slug(
+            desired_slug_base="A mixed CASING str",
+            desired_org_name="santry",
+            owning_user_id=42,
+        )
+        assert slug == "a-mixed-casing-str-9e9173167"
+
+    def test_slug_with_unicode_chars(self):
+        unicoded_str = "Sí Señtry 😅"
```

Comment:
I'm open for more complex strings to test this functionality, but really it's just Django's default unicode slug handling doing the heavy lifting.